### DISCUSSION TITLE
Using the USE_AUTO_VECTOR option.

### DISCUSSION THREAD

**Discussion by Gliese852** [davisking/dlib#2887] [General]
I need to run `dlib` on a machine without `avx` and even `sse4`, but the build, including python module, is done on a machine that _has_ these extensions. While solving the problem, I came across this option, but I had ~~two~~ one question:

1. Based on [this line](https://github.com/davisking/dlib/blob/f7d99ae0dc97c00c1690863881709b7a8b89bb40/dlib/cmake_utils/set_compiler_specific_options.cmake#L31), should I set the `USE_AUTO_VECTOR` to `true` to disable using vector extensions? Shouldn't it be the opposite?

~~2. When building the [pybind11](https://github.com/davisking/dlib/blob/f7d99ae0dc97c00c1690863881709b7a8b89bb40/dlib/cmake_utils/set_compiler_specific_options.cmake#L6) module, this option has no effect, is this intentional?~~

**Comment by davisking** [comment]
Pass in `--no USE_AVX_INSTRUCTIONS` when building the python package and it will not use avx instructions.
